src/com/facebook/buck/apple/project_generator/ProjectGenerator.java

```diff
@@ -1578,6 +1583,46 @@ private PBXNativeTarget generateBinaryTarget(
     return target;
   }
 
+  private void setAppIconSettings(
+      ImmutableSet<AppleAssetCatalogDescriptionArg> recursiveAssetCatalogs,
+      ImmutableSet<AppleAssetCatalogDescriptionArg> directAssetCatalogs,
+      BuildTarget buildTarget,
+      ImmutableMap.Builder<String, String> defaultSettingsBuilder) {
+    ImmutableList<String> appIcon =
+        Stream.concat(directAssetCatalogs.stream(), recursiveAssetCatalogs.stream())
+            .map(x -> x.getAppIcon())
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .collect(ImmutableList.toImmutableList());
+    if (appIcon.size() > 1) {
+      throw new HumanReadableException(
```

Comment:
My only concern here is that we're adding a new constraint which has the potential to break project generation for existing codebases. Changing the behaviour of Buck in this way can be painful when deploying new builds at Facebook (which would lead to reverts and then having to do the whole commit dance again). Instead, I'd suggest one of the following (or a combination):

- Emit a warning rather than throw an exception.
- If we want to keep the ability to throw an exception, we can add a config option for it.

I'm wondering, what happens when Buck itself builds an app that has multiple catalogs with `app_icon` defined?